risc0/circuit/bigint/src/byte_poly.rs

```diff
@@ -96,6 +96,17 @@ impl BytePoly {
         Self::from_biguint(rem, coeffs)
     }
 
+    pub fn nondet_inv<Rhs: Borrow<BytePoly>>(&self, rhs: Rhs, coeffs: usize) -> BytePoly {
+        // Computes the inverse of LHS mod RHS via the `inv = lhs^{rhs - 2} % rhs` algorithm
```

Comment:
Inverse of LHS mod RHS makes sense for coprime LHS and RHS. Is that assumed? If so, INV = LHS^{phi(RHS) - 1} mod RHS where phi is Euler's totient function. INV = LHS^{RHS-2} mod RHS only for prime RHS. Is that assumed too?

Reply:
Good point! I've added clarifying comments to indicate that this requires prime RHS (which implies coprime LHS & RHS as well). We may generalize the algorithm later, but at the moment, I am using this with prime RHS; and this is not a soundness issue, as this is a nondet and the correctness of the inversion must be checked anyway.